### PR TITLE
Creating category and finding all category

### PR DESCRIPTION
**Pull request by TijanAyo**

### CODE REVIEW

File: src/categories/categories.controller.ts [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
 export class CategoriesController {
-  constructor(private readonly categoriesService: CategoriesService) {}
+  constructor(private readonly categoryService: CategoriesService) {}
 
-  @Post()
-  create(@Body() createCategoryDto: CreateCategoryDto) {
-    return this.categoriesService.create(createCategoryDto);
+  @Post('/create')
+  @ApiOperation({summary: 'Create a new category'})
+  async create(@Body() createCategoryDto: CreateCategoryDto) {
```
Comment (Chimexx) [inline src/categories/categories.controller.ts:14]: What is this async function returning? a promise, `Promise<Ok<Category>>` A lot of codes in the dummy core can give you hints, study the way they were written

Also replicate it in other controllers and service files

Comment (TijanAyo) [reply src/categories/categories.controller.ts:14]: Noted... I will return the types properly

File: src/categories/categories.service.ts [hunk 2]
```diff
@@ -1,15 +1,59 @@
...
+      )
+    }
   }
 
-  findAll() {
-    return `This action returns all categories`;
+  async findAllCategory(): Promise<Category[]> {
+    try{
+      const category = await this.categoryRepo.find({
+        select: { id: true, name: true, description: true}
```
Comment (Chimexx) [inline src/categories/categories.service.ts:48]: I don't understand what you did here, simply do `find()` and it returns all entries

File: src/categories/categories.controller.ts (hunk 1 above)
Comment (TijanAyo) [reply src/categories/categories.controller.ts]: I was able to fix a minor bug... Thanks for this

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/categories/categories.service.ts (hunk 2 above)
Comment (TijanAyo) [reply src/categories/categories.service.ts:48]: In the dummy service... they returned specific entries like the id, name and description

Comment (Chimexx) [reply src/categories/categories.service.ts:48]: Understood, what if the frontend need more than those fields? please remove the select filter

**Review by Chimexx — DISMISSED** [review]

File: src/categories/categories.controller.ts [hunk 3]
```diff
@@ -1,34 +1,41 @@
 import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { ApiOperation } from '@nestjs/swagger';
+import { Ok, ZaLaResponse } from '../common/helpers/response';
 import { CategoriesService } from './categories.service';
 import { CreateCategoryDto } from './dto/create-category.dto';
 import { UpdateCategoryDto } from './dto/update-category.dto';
+import { Category } from '../entities/category.entity';
```
Comment (johnayinde) [inline src/categories/categories.controller.ts:8]: i can see the swagger API tag

**Review by johnayinde — CHANGES_REQUESTED** [review]

File: src/categories/categories.controller.ts (hunk 3 above)
Comment (johnayinde) [reply src/categories/categories.controller.ts:8]: @TijanAyo update your branch with the latest changes from the main branch

Comment (TijanAyo) [reply src/categories/categories.controller.ts:8]: Omo I don't know what is wrong with my branch... i pulled before merging

**Review by johnayinde — DISMISSED** [review]

**Comment by PhilemonBrain** [comment]
@TijanAyo ensure this branch is updated. @Chimexx and @Olasunmade please review

File: src/categories/categories.controller.ts [hunk 4]
```diff
@@ -1,44 +1,49 @@
...
   @Get()
-  findAll() {
-    return this.categoriesService.findAll();
+  @ApiOperation({summary: 'Get all available category'})
+  async findAll(): Promise<Ok<Category[]>>{
+    const allCategories = await this.categoryService.findAllCategory();
+    return ZaLaResponse.Ok(allCategories, 'Ok', '200')
   }
 
   @Get('apis:categoryId')
```
Comment (SunmadeBabafemi) [inline src/categories/categories.controller.ts:28]: correct this path.. "apis/:categoryId" so it can be picked as a request param

**Review by SunmadeBabafemi — CHANGES_REQUESTED** [review]

File: src/categories/categories.controller.ts (hunk 4 above)
Comment (TijanAyo) [reply src/categories/categories.controller.ts]: Resolved ...

**Review by SunmadeBabafemi — APPROVED** [review]

**Review by Chimexx — APPROVED** [review]

**Review by johnayinde — APPROVED** [review]

**Pull request merged**